### ISSUE TITLE
[DateRangePicker] How to customize the header of the DateRangePicker or DeskTopDateRangePicker?

### ISSUE THREAD

**Issue by jin0106** [mui/mui-x#11407]
### The problem in depth 🔍

I've thoroughly explored the MUI-X repository, but it seems that the DesktopDateRangePicker component does not have the calendarHeader slot, and there is no apparent functionality to customize the calendar header.

What I want to do is change the date format of the header. For example, I would like to change Dec, 2023 to 2023, Dec.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
System:
    OS: macOS 13.3
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 120.0.6099.71
    Edge: Not Found
    Safari: 16.4
  npmPackages:
    @emotion/react:  11.11.1 
    @emotion/styled:  11.11.0 
    @mui/base:  5.0.0-beta.12 
    @mui/core-downloads-tracker:  5.14.19 
    @mui/icons-material: ^5.14.19 => 5.14.19 
    @mui/material: ^5.14.6 => 5.14.6 
    @mui/private-theming:  5.14.19 
    @mui/styled-engine:  5.14.19 
    @mui/system:  5.14.19 
    @mui/types:  7.2.10 
    @mui/utils:  5.14.19 
    @mui/x-date-pickers: ^6.16.1 => 6.16.1 
    @mui/x-date-pickers-pro: ^6.16.1 => 6.16.1 
    @mui/x-license-pro: ^6.10.2 => 6.10.2 
    @types/react: ^18.0.26 => 18.2.20 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: 4.9.5 => 4.9.5 

```
</details>

**Search keywords**: dateRangePicker customize header
**Order ID**: 58410

**Comment by michelengelen** [comment]
Hey @jin0106 
This adjustment is not possible in `v6`. We are in the middle of preparing `v7` and this is a feature we implemented with #10990. Here is a [demo](https://codesandbox.io/p/sandbox/stupefied-hoover-d2vtv7?file=%2Fsrc%2FDemo.tsx%3A13%2C9) of that.
You can use this feature already when installing the `next` version of the date pickers package:
```shell
yarn add @mui-x/x-date-pickers@next
```

```shell
npm i @mui-x/x-date-pickers@next
```
Please be aware that there might be breaking changes, since we are still in alpha phase.

Since this is already being added as a new feature in `v7` I will close this issue now.

If there is anything else I can help you with feel free to reopen this issue.

Thanks 🙇🏼

**Comment by jin0106** [comment]
@michelengelen 

Thank you for your answer! If there is a schedule for the beta release of v7, when can I expect to find out about it?
